### ISSUE TITLE
Thaumic Horizons Pocket Plane generating in Overworld

### ISSUE THREAD

**Issue by xeronut** [Team-RTG/Realistic-Terrain-Generation#241]
Here is a link to my post on MCF - http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/wip-mods/2524489-wip-realistic-terrain-generation-0-0-12?comment=311

There are no biome conflicts at ID 69, and in Standard/Extreme it does not appear.  Alternate Terrain Generation (ATG) mod also does not use this biome.  As there is no specific compatibility built for Thaumic Horizons, and this plane was never meant to spawn in the Overworld, I'd like to ask for an internal blacklisting for this biome please.  All additional info can be found in the post above.  Thank you for your time!

**Comment by whichonespink44** [comment]
Thanks for the info. There's also some weird behaviour happening with BOP biomes generating even though they've been disabled in the RTG configs, so maybe they're related.

I'm actually working on that issue as we speak, so hopefully solving that one will resolve this one as well.

I'll keep you posted either way.

**Comment by whichonespink44** [comment]
@xeronut Can you do me a favour? Can you remove Thaumic Horizons but keep all the other mods you have installed, then load up a new world, dump the biomes via NEI and attach the CSV dump to this issue? I think I know what's causing it to generate in the overworld, but I need to see the other biomes to be sure.

**Comment by xeronut** [comment]
Here you go - http://pastebin.com/YFVUVGr7

**Comment by scalda** [comment]
Just to let you know i have got the same thing here lol came and found this issue and thought i would post saying i'm getting it aswell

**Comment by whichonespink44** [comment]
Ok, I've spent a few hours on this and still haven't been able to find a Pocket Plane biome generating in the Overworld using the latest dev build.

So unfortunately @xeronut, the only things I can suggest here are... wait until 0.0.14 is released later today, and see if it's still happening, and also try downgrading to the recommended version of Forge, [10.13.4.1448](http://files.minecraftforge.net/maven/net/minecraftforge/forge/index_1.7.10.html) (the one on the right)

I can't really add a blacklist because RTG is already using a whitelist. It only generates biomes that I have specifically added support for. Even in the case of @scalda's Maple Woods ID conflict, it still shouldn't be generating the Pocket Plane because I haven't added support for it. RTG doesn't use IDs - it either uses the API that the biome-adding mod has provided (e.g. [BOP](https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/master/src/main/java/biomesoplenty/api/content/BOPCBiomes.java) or [EBXL](https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/master/src/main/java/extrabiomes/api/BiomeManager.java)) or it checks to see if a supported mod is installed and loops through those biomes and adds support for them on an individual basis (e.g. [Thaumcraft](https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/master/src/main/java/rtg/world/biome/realistic/thaumcraft/RealisticBiomeTCBase.java#L44-L59)).

So the only explanations that I can think of are that some other mod is forcefully 'injecting' the Pocket Plane into the Overworld somehow (which I don't even know how that's possible), or maybe the later version of Forge is doing something similar. I really don't know to be honest.

Sorry that's probably not much help, but maybe at least it will trigger some dialogue between us all and maybe get us closer to figuring out what's going on.

**Comment by xeronut** [comment]
That's cool.  :)  Thanks for doing all this work to pin it down.  THorizons is an awesome mod that unfortunately has been left abandoned before the dev implemented the config file (who *does* that anymore?) and it's just a waiting game until they come back from hiatus to fix the biome behavior.  It was never meant to spawn anywhere but the pocket plane dimensions - what is essentially another dimension generation mod based on the Thaumcraft aspects, but was unfinished - and was included into the last release.  If memory serves me right THorizons was removed from a large modpack recently because it had no method of changing its biome ID and BoP already placed stuff down at ID69, so either they moved the BoP biome (which is logical for new worlds to generate properly, but was a bad move for existing ones) or they removed THorizons.  I'll run 0.0.14 through its paces and see if the Pocket Plane still generates in the Overworld early in the week and leave some feedback here.

**Comment by whichonespink44** [comment]
I can definitely relate to the frustration of an awesome abandoned mod ;)

It's a shame that THorizons isn't open source so that someone else could pick it up and add the config file.

In any case, let me know how you get on with 0.0.14... I think I'm gonna start getting it ready in an hour or so, as there's a lot to write about in the announcement! :)

**Comment by whichonespink44** [comment]
@xeronut - I was going to look at this again tonight, but wanted to check with you first to see if you've had a chance to test the latest version.

If it's still happening, do you have any tips for finding the Pocket Plane in the overworld? Or is it just a case of getting (un)lucky?

**Comment by xeronut** [comment]
It seems unluckiness, really.  I have ExtraBiomesXL (with a couple of the biomes disabled) and Thaumcraft for Overworld biome mods, so I should see the Pocket Plane relatively often.. but no dice now with 0.0.15.  I've made a few worlds this morning and even went looking specifically for snowy biomes (what the Pocket Plane was showing up as in the screens I posted above) and have yet to encounter it.  I even triple-checked to make sure there were no biome conflicts at ID69 - nothing.  I'll make a couple more this morning and edit this post with findings.

**Comment by whichonespink44** [comment]
Cool, I'll do the same. It's possible that 0.0.15 fixed it, though I can't see how it could have. Only one way to find out!! :)

**Comment by whichonespink44** [comment]
I still haven't been able to find the Pocket Plane in the overworld in the latest version, so I think I might close this one @xeronut unless you've been able to find one?

On a more positive note, it looks like TheKentington is back: http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/2361575-thaumic-horizons-v1-1-6-thaumcraft-4-addon?comment=1773

Someone asked about the Pocket Plane ID being configurable in the next release, but I don't think he responded.

**Comment by xeronut** [comment]
Have not found anything in 14 test worlds, two next door to huge snowy biomes.  In those two worlds I was creeping in F3 checking to see if there were spotty Pocket Planes biomes block-wide.. I saw nothing at all.  Closing it sounds like a good idea.  I will keep looking for weirdness however, in case this is more elusive.

Holy crap, son.  Hadn't noticed TheKentington coming back at all.  BRB cleaning brains off the wall.
